### PR TITLE
[23.0] Fix Invocation RO-Crate metadata

### PR DESCRIPTION
**Pull request by pauldg**
Follow up on #15101 to fix bugs in the exported RO-Crate metadata as specified in the issue #15642 and described in more detail here https://github.com/ResearchObject/workflow-run-crate/pull/44

This PR implements the required changes to the RO-Crate metadata and updates to the corresponding unit/api/integration tests.

## How to test the changes?
(Select all options that apply)
- [X] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [X] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by stain** [comment]
Test 3.7 failed because of timeout - ignore

File: lib/galaxy/model/store/ro_crate_utils.py [hunk 1]
```diff
@@ -67,88 +85,148 @@ def __init__(self, model_store: Any):
...
-                filename, _ = self.model_store.dataset_id_to_path[dataset.dataset.id]
-                if not filename:
-                    filename = f"datasets/dataset_{dataset.dataset.uuid}"
-                name = dataset.name
-                encoding_format = dataset.datatype.get_mime()
+    def _add_file(self, dataset: HistoryDatasetAssociation, properties: Dict[Any, Any], crate: ROCrate) -> File:
+        if dataset.dataset.id in self.model_store.dataset_id_to_path:
+            filename, _ = self.model_store.dataset_id_to_path[dataset.dataset.id]
+            if not filename:
+                filename = f"datasets/dataset_{str(dataset.dataset.uuid)}"
```
Comment (davelopez) [inline lib/galaxy/model/store/ro_crate_utils.py]: I would expect the str casting to be redundant here, don't f-strings take care of that directly?

File: test/unit/data/model/test_model_store.py [hunk 2]
```diff
@@ -424,7 +425,7 @@ def validate_other_entities(ro_crate: ROCrate):
     inputs = workflow["input"]
     outputs = workflow["output"]
     assert inputs[0]["additionalType"] == "File"
-    assert inputs[1]["additionalType"] == "None"
+    # assert inputs[1]["additionalType"] == "Text"
```
Comment (davelopez) [inline test/unit/data/model/test_model_store.py]:
```suggestion
```

You can directly remove it if it is not checked anymore

File: test/unit/data/model/test_model_store.py [hunk 3]
```diff
@@ -456,18 +457,22 @@ def validate_invocation_collection_crate_directory(crate_directory):
     actions = [_ for _ in ro_crate.contextual_entities if "CreateAction" in _.type]
     assert len(actions) == 1
     wf_action = actions[0]
-    wf_objects = wf_action["object"]
-    assert len(workflow["input"]) == 3
+    assert wf_action in root["mentions"]
+    # wf_objects = wf_action["object"]
```
Comment (davelopez) [inline test/unit/data/model/test_model_store.py]:
```suggestion
```

Same here :)

**Review by davelopez — APPROVED** [review]
Looks good to me!
Thank you so much @pauldg for adding the missing metadata and increasing the test coverage!

When you think is finally ready for review take it out of draft so we can merge it.

File: lib/galaxy/model/store/ro_crate_utils.py (hunk 1 above)
Comment (pauldg) [reply lib/galaxy/model/store/ro_crate_utils.py]: Yep, that works fine.
```suggestion
                filename = f"datasets/dataset_{dataset.dataset.uuid}"
```

**Comment by mvdbeek** [comment]
Thanks @pauldg!

**Pull request merged**